HEN_HOUSE/egs++/geometry/egs_autoenvelope/egs_autoenvelope.h

```diff
@@ -0,0 +1,589 @@
+
+
+Region Dection / Volume Correction Algorithm
+--------------------------------------------
+
+(Note: Currently the auto envelope can only accomodate a single type of inscribed
+geometry, although it could be adapted to handle an aribtrary number of
+different inscribed geometries.)
+
+With reference to the diagram below you can show that
+if the bounding shape volume is \f$V_{BS}\f$, then the
+inscribed geometry  volume, \f$V_{inscribed}\f$, is estimated as:
+
+\f[
+
+ V_{inscribed} =(total\_in\_inscribed / n\_total\_points) * V_{BS}
+
+\f]
+
+and the volume of the shape inside a give region \f$i\f$, \f$V_{inscribed}(i)\f$ is given by:
+
+\f[
+ V_{inscribed}(i) = (n\_inscribed(i)/n\_total\_points) * V_{BS}
+\f]
+
+So the corrected volume of region \f$i\f$ is just:
+
+\f[
+ V_{corrected}(i) = V_{uncorrected}(i) - (n\_inscribed(i)/n\_total\_points) * V_{BS}
+\f]
+
+
+\image html vc_diagram800.png "Volume correction diagram"
+
+
+The way the algorithm works is roughly:
+
+- Choose a random point in space within the bounding shape.
+- Check whether the point lands within the inscribed geometry
+- If the point is outside, stop and go to step 1
+- If the point is inside increment total number of points falling within
+  inscribe geometry
+- Transform the point to all locations of the inscribed geometry
+- find the region of the base geometry that the point falls within
+- increment the number of hits in that region
+- add the inscribed geometry to the set of all geometries in that region
+
+which in pseudo-code would look something like:
+\verbatim
+
+total_in_inscribed = 0
+
+for i = 1 to n_base_regions
+    n_inscribed[i] =  0
+    geoms_in_reg[i] = empty_set()
+
+for step = 1 to n_total_points
+
+    point  =  get_random_point_within_bounding_shape()
+
+    if inscribed_geom.point_is_inside(point):
+
+        total_in_inscribed += 1
+
+        for inscribed_copy in all_inscribed_copies:
+            transformed_point = transform_point_to_location(inscribed_copy.location, point)
+            base_geom_reg =  base_geom.whichRegion(transformed_point)
+            n_inscribed[base_geom_reg] += 1
+            geoms_in_reg[base_geom_reg].add_to_set(inscribed_copy)
+
+
+\endverbatim
+
+Geometry Definition Format
+--------------------------
+
+There is an example geometry included in the egs_autoenvelope/6702.geom file.
+
+\verbatim
+
+:start geometry:
+
+    library = egs_autoenvelope
+    name = autoenvelope
+    type = EGS_AEnvelope # optional: EGS_AEnvelope(default) or EGS_ASwitchedEnvelope
+    print debug info = no  # optional: no(default), yes
+    output volume correction file = gzip # optional no(default), yes(equivalent to text), text or gzip
+
+    base geometry = name_of_base_geom # the name of a previously defined geometry
+
+    :start inscribed geometry:
+
+        inscribed geometry name = seed # the name of a previously defined geometry
+
+        output volume correction file = text # (optional) text or gzip
```

Comment:
I think that using `output volume correction file` here does nothing, so remove it from the example.